File: java/src/test/java/challenges/aoc2024/day3/Application.java
package challenges.aoc2024.day3;

import challenges.aoc2024.day3.service.DayThreeSolver;
import challenges.aoc2024.day3.service.ProgramService;
import service.TaskSolver;

public class Application {
    public static void main(String[] args) {
        final TaskSolver solver = new TaskSolver(new DayThreeSolver(new ProgramService()));
        solver.solveTasks();
    }
}

File: scala/src/main/scala/challenges/aoc2024/day3/Application.scala
package challenges.aoc2024.day3

import _root_.service.TaskSolver
import challenges.aoc2024.day3.service.{DayThreeSolver, ProgramService}

object Application {
  def main(args: Array[String]): Unit = {
    val solver: TaskSolver = TaskSolver(DayThreeSolver(ProgramService()))
    solver.solveTasks()
  }
}
